shared/reducers/chat.js

```diff
@@ -250,16 +277,7 @@ function reducer (state: Constants.State = initialState, action: Constants.Actio
     }
     case 'chat:updateMessage': {
       const {messageID, message, conversationIDKey} = action.payload
-      // $FlowIssue
-      return state.update('conversationStates', conversationStates => updateConversationMessage(
-        conversationStates,
-        conversationIDKey,
-        item => !!item.messageID && item.messageID === messageID,
-        m => ({
-          ...m,
-          ...message,
-        })
-      )).set('messageMap', state.get('messageMap').set(message.key, message))
```

Comment:
This was buggy -- it doesn't merge the message properties in the messageMap. Using `updateStateWithMessageChanged` fixes this.